addons/account/models/account_bank_statement.py

```diff
@@ -170,303 +54,208 @@ def _get_move_line_count(self):
+                # we do not change the state of a posted statement automatically
+                continue
+            elif statement.currency_id and statement.currency_id.compare_amounts(
+                    statement.balance_end,
+                    statement.balance_end_real) == 0:
+                statement.state = 'complete'
+            else:
+                statement.state = 'open'
+
+    def _compute_attachment_ids(self):
+        for statement in self:
+            attachment_ids = self.env['ir.attachment'].search([
+                ('res_id', '=', statement.id),
+                ('res_model', '=', 'account.bank.statement')
+            ]).ids
+            message_attachment_ids = statement.mapped('message_ids.attachment_ids').ids  # from mail_thread
+            statement.attachment_ids = [Command.set(list(set(attachment_ids) - set(message_attachment_ids)))]
+
+    def _inverse_attachment_ids(self):
+        # remove the attachments deleted in the form
+        # the many2many_binary widget creates the attachments automatically, but it doesn't remove them when deleted
+        self.env['ir.attachment'].search([
+                ('res_id', 'in', self.ids),
+                ('res_model', '=', self._name),
+                ('id', 'not in', self.attachment_ids.ids)
+            ]).unlink()
+
+    name = fields.Char(
+        string='Reference',
+        states={'posted': [('readonly', True)]},
+        copy=False,
+        index=True,
+    )
+    date = fields.Date(compute='_compute_date', store=True, index=True)
+
+    # Balance start  has to be encoded by the user, but the default value is the last completed or posted
+    # statement's real end balance.
+    balance_start = fields.Monetary(string='Starting Balance')
+    balance_end_real = fields.Monetary('Ending Balance', states={'posted': [('readonly', True)]},)
+    state = fields.Selection(
+        string='Status',
+        selection=[
+            ('open', 'Open'),
+            ('complete', 'Complete'),
+            ('posted', 'Posted'),
+        ],
+        compute='_compute_state',
+        store=True,
+        help="""The current state of your bank statement:
+             - Open: Ending balance does not match.
+             - Complete: Ending balance matches the lines, ready for the reconciliation.
+             - Posted: validated and locked."""
+    )
     currency_id = fields.Many2one('res.currency', compute='_compute_currency', string="Currency")
-    journal_id = fields.Many2one('account.journal', string='Journal', required=True, states={'confirm': [('readonly', True)]}, default=_default_journal, check_company=True)
-    journal_type = fields.Selection(related='journal_id.type') # used for usability purposes
+    journal_id = fields.Many2one(
+        comodel_name='account.journal',
+        compute='_compute_journal_id', store=True, readonly=False,
+        # readonly is set to false so we can have a default value in the classic from view
```

Comment:
How does it work if you have lines coming from different journals?
Or if the statement has a different journal than all the lines?

Shouldn't there be an inverse, and a constraint?

Reply:
it was because we had two views with two different flows, now it is solved and it is readonly